src/scripting/lua_map.cc

```diff
@@ -2107,17 +2121,26 @@ MapObjectDescription
 
 .. class:: MapObjectDescription
 
-   A static description of a tribe's map object, so it can be used in help files
-   without having to access an actual object on the map.
-   This class contains the properties that are common to all map objects such as buildings or wares.
+   A static description of a tribe’s object, so it can be used without
+   having to access an actual object on the map. This class contains the
+   properties that are common to all objects a tribe has, such as buildings or
+   wares.
+
+   The dynamic class :class:`MapObject` corresponding to this class is the base
+   class for all Objects in widelands, including immovables and Bobs. This
```

Comment:
```suggestion
   class for all objects in Widelands, including immovables and bobs. This
```

I don't understand the term "dynamic class"

Reply:
The [original text](https://www.widelands.org/documentation/autogen_wl_map/#wl.map.MapObjectDescription) was:

> The dynamic MapObject class corresponding to this class is the base class for all Objects in widelands, including immovables and Bobs. This class can’t be instantiated directly, but provides the base for all others.

Please suggest a better rephrasing.

Reply:
Let's remove the "dynamic" and then the sentence is fine.

"Widelands" should always be Uppercase as it's a name.

A few lines above I dislike the phrasing "description of a **tribe**’s object" – critters, rocks, trees etc are also map objects. Perhaps rephrase as "A static description of an object on the map, so it can be used without having to access an actual instance of an object on the map."

Reply:
What i do not understand is: If MapObject is the base class of all objects, isn't MapObjectDescription a subclass of MapObject? Shouldn't there be a statement like: "Child of: MapObject"?

Reply:
No, there's no inheritance relation between MapObject and MapObjectDescr. We have one inheritance tree for actual instances of objects on the map and one congruent tree for the description of such objects. 
```
         MapObject
        /         \
       Bob         BaseImmovable
   /    |    \           |     \
Ship Critter Worker Immovable PlayerImmovable
               |              /        |     \
             Carrier      PortDock  Building  RoadBase
               |                   /     |…        |  \
             Ferry   ProductionSite Warehouse    Road  Waterway
```
And the same tree *again*, independent from it, with `Description` appended to every name.

Reply:
Such an (graphical) overview of all classes is a thing that i really miss in the whole documentation. For years i have tried to make such a graph, but i gave up after some time.

I have rephrased the text. If my rephrasing is not correct, please feel free to correct it.